sentence_transformers/losses/CachedGISTEmbedLoss.py

```diff
@@ -230,14 +230,15 @@ def embed_minibatch_iter(
     def calculate_loss_and_cache_gradients(self, reps: list[list[Tensor]], reps_guided: list[list[Tensor]]) -> Tensor:
         """Generalized function to calculate the cross-entropy loss and cache the gradients wrt. the embeddings."""
         loss = self.calculate_loss(reps, reps_guided)
```

Comment:
```suggestion
        loss = self.calculate_loss(reps, reps_guided, with_backward=True)
```
I believe this is missing for CGIST.